### PR TITLE
gui-docker: Update to nvidia-docker version 2

### PR DESCRIPTION
**Pull request by rhaschke**
This is a replacement for #1495, updating the gui-docker script to work with [`nvidia-docker 2`](https://github.com/NVIDIA/nvidia-docker/wiki) as well. Both versions, 1 and 2, of nvidia-docker are supported now.
I also added support for DRI-based cards.

### CODE REVIEW

**Comment by rhaschke** [comment]
As pointed out by @felixvd in https://github.com/ros-planning/moveit/pull/1495#issuecomment-508632232, it would be comfortable for unexperienced docker users to just run gui-docker without the need to remember options to pass. Also, support for persistent containers would be great, i.e. the ability to re-enter a previously ran container.
I added support for both features now: `gui-docker -c <container>` (or just `gui-docker -c`) allows to continue an existing container with the given name (or default_container).
If the container doesn't yet exist and no (positional) options are provided to specify the docker image and command to execute, a meaningful default is used.

Note: Continuing an existing container doesn't stop the container in the end. I like this feature, because it allows to run an additional process within the same container. But there might be other opinions on that.

**Comment by BryceStevenWilley** [comment]
I'm interested in trying this out, I'll look at it tonight.

File: .docker/gui-docker [hunk 1]
```diff
@@ -85,6 +103,10 @@ if [ -n "${CONTAINER_NAME:-}" ] ; then
    if [ -z "$($DOCKER_EXECUTABLE ps -aq --filter name=^$CONTAINER_NAME\$)" ] ; then
       # container not yet existing, add an option to name the container when running docker below
       NAME_OPTION="--name=$CONTAINER_NAME"
+      if [ $(count_positional_args) -eq 0 ] ; then
+         # If no further (positional) arguments were provided, start a bash in the default image (for dummy users)
```
Comment (felixvd) [inline .docker/gui-docker]: > dummy users

Harsh, harsh.

**Comment by felixvd** [comment]
> I added support for both features now: `gui-docker -c <container>` (or just `gui-docker -c`) allows to continue an existing container with the given name (or default_container).

Great. I would go even further and make that the default behavior when `gui-docker` is called without arguments.

> Note: Continuing an existing container doesn't stop the container in the end. I like this feature, because it allows to run an additional process within the same container. But there might be other opinions on that.

I think this is fine and will test soon.

**Comment by rhaschke** [comment]
> I would go even further and make that the default behavior when gui-docker is called without arguments.

This cannot become the default, as it requires to specify a container name.

**Comment by felixvd** [comment]
I assumed it would start up "default_container" as you suggested, unless I misunderstood what you meant.

**Comment by rhaschke** [comment]
Indeed, if there are no arguments at all, we could go for `default_container`.

**Comment by rhaschke** [comment]
Done

File: .docker/gui-docker [hunk 2]
```diff
@@ -1,56 +1,139 @@
...
+function check_dri() {
+    # If there is no /dev/dri, bail out
+    test -d /dev/dri || return 1
+    DOCKER_PARAMS="--device=/dev/dri --group-add video"
+}
+
+function transfer_x11_permissions() {
+    # store X11 access rights in temp file to be passed into docker container
+    XAUTH=/tmp/.docker.xauth
+    touch $XAUTH
```
Comment (BryceStevenWilley) [inline .docker/gui-docker]:
```suggestion
    touch "$XAUTH"
```
I'll default to you, but my understanding of bash scripts is that all variables should be in quotes.

Comment (rhaschke) [reply .docker/gui-docker]: Indeed, it's good practice to quote variables. In this case, it's not necessary, because the variable doesn't contain spaces.

**Pull request merged**